masterbase/app.py

```diff
@@ -268,8 +270,18 @@ def provision(request: Request) -> Redirect:
     """
     # enforce https on base_url
     base_url = str(request.base_url)
-    if not base_url.startswith("https") and not os.environ["DEVELOPMENT"]:
-        base_url = base_url.replace("http", "https")
+    dev_mode = os.getenv("DEVELOPMENT")
+    if dev_mode is not None and dev_mode.lower() == "true":
+        if base_url.startswith("https://"):
+            base_url = base_url.replace("https://", "http://")
+        else:
+            base_url = "http://" + base_url
+    else:
+        if base_url.startswith("http://"):
+            base_url = base_url.replace("http://", "https://")
+        else:
+            base_url = "https://" + base_url
```

Comment:
nice, maybe worth making this a function somewhere else and calling here

Reply:
This function seems a bit buggy..
If dev_mode is enabled and the base_url starts with http://, it will get  modified to be `http://http://<base url>`.
If the base_url starts with https://, then it will be set to `http://<base url>` as expected.
Same goes for the branch when dev_mode is disabled (except with https:// instead).

You can also simplify the actual string replacement, since calling replace on "https://" on a string with "http://" is just a no-op and returns the original string.

```py
# Default a protocol if the base_url doesn't start with one
if not base_url.startswith("http"):
    base_url = "https://" + base_url

if os.getenv('DEVELOPMENT', 'false').lower() == 'true':
    base_url = base_url.replace("https://", "http://") 
else:
    base_url = base_url.replace("http://", "https://") 
```

Reply:
Alternatively, you can do some pythonic splitting...
```py
proto = "https://" if os.getenv('DEVELOPMENT', 'false').lower() == 'true' else "http://"
base_url = proto + base_url.split("//")[-1]
```